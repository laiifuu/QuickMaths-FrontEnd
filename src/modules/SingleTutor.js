import { Link, useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { FaFacebookF, FaInstagram, FaTwitter } from 'react-icons/fa';
import { IoChevronForwardCircleOutline } from 'react-icons/io5';

const SingleTutor = () => {
  const { id } = useParams();
  const tutor = useSelector((state) => state.tutors.tutors).find(
    (item) => item.id === Number(id),
  );
  const state = useSelector((state) => state.tutors.tutors);

  if (!tutor && state.length === 0) {
    return <div className="loading">Loading</div>;
  }
  if (!tutor && state.length !== 0) {
    return <div className="loading">Element not found</div>;
  }
  return (
    <section className="tutor-details-page">
      <div className="tutor-photo-container">
        <img
          src={tutor.photoUrl}
          alt="tutor"
          className="detailsPageTutorPhoto"
        />
      </div>
      <div className="tutor-details-container">
        <h1>
          {tutor.firstName}
          {' '}
          {tutor.lastName}
        </h1>
        <p className="tutor-description">
          {tutor.description}
        </p>
        <ul className="details">
          <li>
            <span>Hourly fee: </span>
            <span>
              {tutor.hourlyFee}
              $
            </span>
          </li>
          <li>
            <span>Years of experience: </span>
            <span>{tutor.experience}</span>
          </li>
          <li>
            <span>Rating: </span>
            <span>{tutor.rating}</span>
          </li>
        </ul>
        <ul className="tutor-social-links">
          <li>
            <a
              href={
                tutor.fbLink !== 'blank' ? tutor.fbLink : 'https://facebook.com'
              }
              className="tutor-sm-link"
              target="_blank"
              rel="noreferrer"
              aria-label="facebook page"
            >
              <FaFacebookF />
            </a>
          </li>
          <li>
            <a
              href={
                tutor.igLink !== 'blank' ? tutor.igLink : 'https://instagram.com'
              }
              className="tutor-sm-link"
              target="_blank"
              rel="noreferrer"
              aria-label="instagram page"
            >
              <FaInstagram />
            </a>
          </li>
          <li>
            <a
              href={
                tutor.twitterLink !== 'blank'
                  ? tutor.twitterLink
                  : 'https://twitter.com'
              }
              className="tutor-sm-link"
              target="_blank"
              rel="noreferrer"
              aria-label="twitter page"
            >
              <FaTwitter />
            </a>
          </li>
        </ul>
        <Link
          className="makeReservationButton"
          to="/reserve"
          state={{ chosenTutorId: tutor.id }}
        >
          <button type="button">
            Make reservation
            <IoChevronForwardCircleOutline className="reserve-arrow-icon" />

          </button>
        </Link>
      </div>
    </section>
  );
};

export default SingleTutor;
