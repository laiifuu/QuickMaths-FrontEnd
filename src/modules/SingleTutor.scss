.detailsPageContainer {
  display: grid;
}

.detailsPageTutorPhoto {
  margin: 1rem auto;
}

.detailsPageTutorDescription {
  padding: 1rem 1rem 0 1rem;
  text-align: center;
}

.text-center {
  text-align: center;
}

.makeReservationButton {
  padding: 1rem 2rem;
  background-color: green;
  border-radius: 2rem;
  display: block;
  text-align: center;
  color: white;
  max-width: 300px;
  margin: 1rem auto;
}

.font-2rem {
  font-size: 2rem;
}

.line-height15 {
  line-height: 1.5;
}

.font-gray {
  color: rgb(51, 51, 51);
}

.detailsPageGrid {
  display: grid;
  grid-template-rows: 1fr 1fr 1fr;
  padding: 0 1rem;

  li {
    padding: 0.3rem;
    display: flex;
    justify-content: space-between;

    span {
      margin-right: auto;
    }
  }
}

.loading {
  width: 100%;
  height: 100%;
  text-align: center;
  padding-top: 50vh;
}

@media only screen and (min-width: 768px) {
  .detailsPageContainer {
    padding: 2rem;
    display: grid;
    width: calc(100% - 10rem);
  }

  .detailsPageTutorPhoto {
    margin: auto auto;
  }
}

@media only screen and (min-width: 1100px) {
  .detailsPageContainer {
    padding: 2rem;
    display: grid;
    grid-template-columns: 1fr 360px;
    width: calc(100% - 10rem);
  }

  .detailsPageTutorPhoto {
    margin: auto auto;
  }
}
