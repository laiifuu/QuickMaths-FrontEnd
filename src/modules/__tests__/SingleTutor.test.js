import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/configureStore';
import SingleTutor from '../SingleTutor';

describe('Show one tutor', () => {
  test('Show redirection', async () => {
    const singleTutor = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <SingleTutor />
        </BrowserRouter>
        ,
      </Provider>,
    )
      .toJSON();
    expect(singleTutor).toMatchSnapshot();
  });
});
