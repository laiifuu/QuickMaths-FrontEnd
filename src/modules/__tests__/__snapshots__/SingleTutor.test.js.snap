// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show one tutor Show redirection 1`] = `
Array [
  <div
    className="loading"
  >
    Loading
  </div>,
  ",",
]
`;
